src/import_edf.jl

```diff
@@ -365,44 +365,59 @@ and an `AmbiguousChannelError` displayed as a warning.
 
 See the OndaEDF README for additional details regarding EDF formatting expectations.
 """
-function store_edf_as_onda(path, edf::EDF.File, uuid::UUID=uuid4();
+function store_edf_as_onda(edf::EDF.File, onda_dir, recording_uuid::UUID=uuid4();
                            custom_extractors=STANDARD_EXTRACTORS, import_annotations::Bool=true,
                            postprocess_samples=identity,
                            signals_prefix="edf", annotations_prefix=signals_prefix)
+
+    # Validate input argument early on
+    signals_path = joinpath(onda_dir, "$(validate_arrow_prefix(signals_prefix)).onda.signals.arrow")
+    annotations_path = joinpath(onda_dir, "$(validate_arrow_prefix(annotations_prefix)).onda.annotations.arrow")
+
     EDF.read!(edf)
     file_format = "lpcm.zst"
 
-    signals = Any[]
+    signals = Onda.Signal[]
```

Comment:
> What's the advantage of specifying the narrow type here?

1. Consistency: The returned `annotations` already is returns a `Vector{Onda.Annotation}` so having `signals` return a `Vector{Onda.Signal}` keeps things consistent
2. Dispatch: returning an `Onda.Signal` type means that callers of `store_edf_as_onda` can use this type during dispatch

>  I thought the design philosophy of Onda was to leave the validation to the reading/writing part and not rely on the structs since those may change...

If you have a reference for this that would be great.

Reply:
I think it's fine for a direct constructors of rows to type the rows however they want, if there's a reason to do so; "consumers" of rows should generally not type-restrict the rows they accept, however.

in this case, I don't see the reason for doing this, it seems redundant

this is actually a general part of the Legolas/Tables philosophy (and follows the YAS philosophy as well of preferring not to defensively restrict dispatch). i think it is written/discussed in a few places but i'll try to dig up a good reference or file an issue to create on otherwise

> Dispatch: returning an Onda.Signal type means that callers of store_edf_as_onda can use this type during dispatch

Nothing consumers from dispatching on the types of the actual values even if the container doesn't have a concrete eltype. Unless you mean dispatch on the container's eltype? IMO that's generally bad practice for generic code and should only be done for performance optimization (not for "behavioral" dispatch)

Reply:
Should `edf_to_onda_annotations` also be changed to return a `Vector{Any}` instead of returning a `Vector{Annotation}`?

Overall I have some concerns about this approach being harder to reason about but I do see the advantages. Thanks for the YAS mention as that documentation cleared up some of the intent for me.

I did look a little more into this and discovered, what I believe to be, a flaw in Onda.jl which makes this change no longer necessary.

Reply:
🚚 *beep* *beep*. Going to need to back up a little bit

> Dispatch: returning an `Onda.Signal` type means that callers of `store_edf_as_onda` can use this type during dispatch

When I mentioned this point I was actually referring to why I changed the code below into explicitly constructing an `Onda.Signal` and not that I want the eltype to be something specific. I agree that consumers of this shouldn't be relying on dispatching on the eltype.

> I did look a little more into this and discovered, what I believe to be, a flaw in Onda.jl which makes this change no longer necessary.

I did not find a bug in Onda.jl. Originally, I was thinking that `Onda.Signal`'s `file_path` should enforced to be a string which is not the case. I thought this initially because of `onda.signal@1` [stating that `file_path` is of type `Utf8`](https://github.com/beacon-biosignals/Onda.jl#ondasignal1).

Looking into this revealed that [`Onda.store` is already returning a `Onda.Signal`](https://github.com/beacon-biosignals/Onda.jl/blob/54453452d6a2a619da19cde97cf423cba2330f88/src/samples.jl#L503) so the actual problem is that we're forcing `file_path` here to always be a `String`:

https://github.com/beacon-biosignals/OndaEDF.jl/blob/775c1ce56bd0ba0b36b79e334b319e9044d0f1fb/src/import_edf.jl#L383

Using `rowmerge` ends up returning a `NamedTuple` and not a `Onda.Signal` as provided from `store`. By modifying the code to be:

```julia
signal = store(file_path, file_format, samples, uuid, Second(0))
```

We get an `Onda.Signal` returned with the `file_path` property being whatever was passed into `store`. I think this is a definite improvement.

As for narrowing the type on the array I think having `Onda.Signal[]` protects against code modifications like `rowmerge` where suddenly the row type differs from what is expected. It's definitely not necessary (especially for dispatch) but I do think it makes the code easier to understand. I would like to have this function be internally consistent as currently we also have:

https://github.com/beacon-biosignals/OndaEDF.jl/blob/775c1ce56bd0ba0b36b79e334b319e9044d0f1fb/src/import_edf.jl#L398

Reply:
I added a test to Onda.jl because of the above: https://github.com/beacon-biosignals/Onda.jl/pull/109

Reply:
@jrevels I've left in `Onda.Signal` type restriction here only because it was useful in catching the issue with `rowmerge` not returning the expected type. Let me know what you think so we can proceed

Reply:
> rowmerge not returning the expected type

this is framed as an issue but I don't think it is, I wouldn't "expect" `rowmerge` to return an `Onda.Signal`, just to return a valid Tables.jl row 

in other words:

> signal = store(file_path, file_format, samples, uuid, Second(0))

I do agree this code is better, but only because it's cleaner/preserves `file_path`'s type better, not because the resulting `signal isa Signal`. No inherent value to the latter, IMO. 

anyway, definitely discussing minutiae at this point so probably fine to just move on as-is